src/components/Layout/Header.component.jsx

```diff
@@ -0,0 +1,147 @@
+    `;
+
+    const NavHeader = styled.div `
+        padding: 4px 20px;
+        width: 100%;
+        display: flex;
+        align-items:center;
+        justify-content: space-between;
+        margin: 0 auto;
+        `;
+
+    const NavLeft = styled.div `
+        display: flex;
+        align-items:center;
+        width:400px;
+        @media (min-width: 768px) {
+        width:40%;
+        }
+    `;
+
+    const SearchInput = styled.div `
+        width:100%;
+        margin-left:20px;
+        height:34px;
+        border-radius: 3px;
+        display:flex;
+        align-items:center;
+        color: #FFF;
+        font-size: 14px;
+        font-weight: 300;
+        background-color: rgba(255, 255, 255, 0.15);
+    `;
+
+    const Input = styled.input `
+        font-size: 16px;
+        border:none;
+        height:1.8em;
+        background-color:transparent;
+        :active,
+        &:focus {
+        outline: none;
+        box-shadow: none;
+        caret-color:white;
+        }
+        ::placeholder,
+        ::-webkit-input-placeholder {
+        color: rgba(255, 255, 255, 0.5);
+
+        }
+        :-ms-input-placeholder {
+        color: rgba(255, 255, 255, 0.5);
+
+        }
+    `;
+
+    const NavRight = styled.div `
+        align-items:center;
+        display:none;
+        @media (min-width: 768px) {
+        display:flex;
+        }
+        overflow:hidden;
+        right:0;
+        svg {
+        margin: 0 10px;
+        }
+
+    `;
+
+    const MenuLink = styled.a `
+    color:white;`;
+
+function Header() {
+
+    
+
+return (
+<div>
+    <Nav>
+        <NavHeader>
+            
+            <NavLeft>
+                <MenuLink href="#">
+                    <Menu fill="currentColor" viewBox="0 0 24 24" aria-hidden="true">
+                        <path d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"></path>
+                    </Menu>
+                </MenuLink>
+                <SearchInput>
+                    <Mag width="24" height="24" fill="none">
+                        <path d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z" stroke="currentColor" strokeWidth="2"
+                            strokeLinecap="round" strokeLinejoin="round"></path>
+                    </Mag>
+                    <Input type="text" placeholder="Search" />
+                </SearchInput>
+            </NavLeft>
+
+            <NavRight>
+                <MenuLink href="#">
+                    <SwitchButton text="Dark Mode" />
+                </MenuLink>
+                <MenuLink href="#">
+                    <Avatar viewBox="0 0 24 24" fill="currentColor" aria-hidden="true">
+                        <path
+                            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z">
+                        </path>
+                    </Avatar>
+                </MenuLink>
+            </NavRight>
+        </NavHeader>
```

Comment:
If you want, you can go further in making more atomic components, so that in the future it can be easier to test each of them depending of the functionality on each of them. Instead of putting all the tests into the header component.

Reply:
I realize that when I was adding the unit testing. It was too late. thanks for the advice! Next time I will do that.